src/test/java/io/confluent/idesidecar/restapi/util/WebClientFactoryTest.java

```diff
@@ -64,41 +64,29 @@ public void webClientOptionsShouldContainUserAgentOnCreation() {
                 )
     );
 
-    // Create a WebClient instance with the correct port
-    WebClient webClient = WebClient.create(
-        Vertx.vertx(), new WebClientOptions()
-            .setDefaultHost("localhost")
-            .setUserAgent("Confluent-for-VSCode/v%s (https://confluent.io; support@confluent.io) sidecar/v%s (%s/%s)"
-                .formatted(sidecarInfo.vsCode(),sidecarInfo.version(),sidecarInfo.osType(), sidecarInfo.osArch()))
-            .setDefaultPort(wireMockServer.port()));
+    String userAgent = webClientFactory.getDefaultWebClientOptions().getUserAgent();
```

Comment:
Maybe we could also test that this is equivalent to `SidecarInfo.getUserAgent()`??

Reply:
We should assert this as Randall mentions. 

We could possibly pass in a different user agent to the WebClient and we'd want this test to fail if that happens so we can investigate to see if that's the desired behavior.